src/value.rs

```diff
@@ -45,10 +52,57 @@ impl ValueType {
 /// `Counter`, `Gauge`, and `Untyped`.
 pub struct Value {
     pub desc: Desc,
-    // TODO: like prometheus client go, use atomic u64.
-    pub val: RwLock<f64>,
     pub val_type: ValueType,
     pub label_pairs: Vec<LabelPair>,
+
+    #[cfg(not(feature = "nightly"))]
+    pub val: RwLock<f64>,
```

Comment:
wrap all in AtomicF64, not in value.
Histogram will use AtomicF64 too. It is not good to use RwLock and AtomicF64 in two places and we have too many nightly cfg flags.